### PR TITLE
refactor: refactor for unit test

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Refactor:
- refactor IPluginCommand to support dependency injection for unit test
- refactor KeyVaultClient to implement IKeyVaultClient for unit test

Test:
- added unit test cases
 
Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

File: Notation.Plugin.AzureKeyVault.Tests/KeyVault/KeyVaultClientTests.cs [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+using Azure;
+using Azure.Core;
+using Azure.Security.KeyVault.Certificates;
+using Azure.Security.KeyVault.Keys;
+using Azure.Security.KeyVault.Keys.Cryptography;
+using Azure.Security.KeyVault.Secrets;
+using Moq;
+using Xunit;
+using Notation.Plugin.Protocol;
+using System.Text;
```
Comment (shizhMSFT) [inline Notation.Plugin.AzureKeyVault.Tests/KeyVault/KeyVaultClientTests.cs:15]: nit: it would be better to sort those imports.

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: Notation.Plugin.AzureKeyVault.Tests/KeyVault/KeyVaultClientTests.cs (hunk 1 above)
Comment (JeyJeyGao) [reply Notation.Plugin.AzureKeyVault.Tests/KeyVault/KeyVaultClientTests.cs]: Updated.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**